app/routes/github.js

```diff
@@ -8,17 +8,29 @@ const {
 } = Ember;
 
 const CODE_INVALID = 'The GitHub authorization code is invalid.';
+const STATE_INVALID = 'Something went wrong in the connect process. Please try again';
 
 export default Route.extend(AuthenticatedRouteMixin, {
-  queryParams: { code: { refreshModel: true } },
+  queryParams: {
+    code: { refreshModel: true },
+    state: { refreshModel: true }
+  },
 
   ajax: service(),
   currentUser: service(),
+  githubState: service(),
   flashMessages:service(),
   store: service(),
 
-  model({ code }) {
-    return this._sendRequest(code);
+  model({ code, state }) {
+    let stateValidator = get(this, 'githubState');
+
+    if (stateValidator.validate(state)) {
+      return this._sendRequest(code);
+    } else {
+      get(this, 'flashMessages').clearMessages().danger(STATE_INVALID);
+      return this.transitionTo('projects-list');
```

Comment:
Why are we transitioning to `projects-list` here? Is there a better place to transition to?